nexus/src/authz/omicron.polar

```diff
@@ -320,25 +321,11 @@ resource Database {
 	    # other general functions.
 	    "modify"
 	];
-	roles = [
-	    # All authenticated users get the "user" role, which grants the
-	    # "query" permission.  See above.
-	    "user",
-
-	    # The special "db-init" user gets the "init" role, which grants the
-	    # additional "modify" permission.
-	    "init"
-	];
-
-	# See above.
-	"query" if "user";
-
-	"user" if "init";
-	"modify" if "init";
 }
 
-# All authenticated users have the "user" role on the database.
-has_role(_actor: AuthenticatedActor, "user", _resource: Database);
+# All authenticated users have the "query" permission on the database.
+has_permission(_actor: AuthenticatedActor, "query", _resource: Database);
+
 # The "db-init" user is the only one with the "init" role.
-has_role(actor: AuthenticatedActor, "init", _resource: Database)
+has_permission(actor: AuthenticatedActor, "modify", _resource: Database)
 	if actor = USER_DB_INIT;
```

Comment:
Untested, but should work:
```suggestion
has_permission(USER_DB_INIT, "modify", _resource: Database);
```

Reply:
Neat!  I'll try that in my follow-up where I'm adding some policy tests.

Reply:
Interestingly, that gave me:

```
thread 'authz::policy_test::test_iam_roles' panicked at 'initializing Oso: loading Polar (Oso) config

Caused by:
    Invalid rule: has_permission(USER_DB_INIT, "modify", _resource: Database{}); Must match one of the following rule types:
    
    has_permission(actor: Actor{}, _permission: String{}, resource: Resource{});
    	Failed to match because: Parameter `USER_DB_INIT` expects a Actor type constraint.
    
    	USER_DB_INIT: Actor
     at line 331, column 1:
    	331: has_permission(USER_DB_INIT, "modify", _resource: Database);
    	     ^
    ', nexus/src/authz/context.rs:33:54
```

but if I make it:

```
has_permission(USER_DB_INIT: AuthenticatedActor, "modify", _resource: Database);
```

that seems to work.